File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.js
function solution(my_string) {    
    const op = my_string.split(" ").filter((x) => isNaN(x));
    const num = my_string.split(" ").filter((x) => !isNaN(x)).map((x) => +x);
    
    return num.reduce((a, b, i) => op[i-1]==="+" ? a+b : a-b);
}

File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.py
def solution(my_string):
    lst = my_string.split()
    answer = int(lst[0])

    for i in range(2, len(lst), 2):
        if lst[i - 1] == '+':
            answer += int(lst[i])
        elif lst[i - 1] == '-':
            answer -= int(lst[i])

    return answer